### DISCUSSION TITLE
What is `path`?

### DISCUSSION THREAD

**Discussion by otakutyrant** [nushell/nushell#10203] [Q&A] [answered]
The book says that we can define the shape of an argument of an extern command as `path` to define the corresponding complention. However `path` seems to be a command rather than a type.

**Comment by amtoine** [comment, marked as answer]
@otakutyrant 
`path` is both a command and a type, i.e. you can do
```nushell
def does-path-exist [somewhere: path]: nothing -> bool {
    $somewhere | path exists
}
```
- the first `path` is the expected type of the `somewhere` argument
- the second `path` is the built-in command and we use the `exists` subcommand

**Reply by otakutyrant** [reply]
Thank you.
